base_layer/core/src/mempool/mempool.rs

```diff
@@ -0,0 +1,149 @@
+        // Pending Pool
+
+        // TODO: Utxs that have been received, verified and have passed all checks, don't have time-locks and only spend
+        // valid UTXOs should be added to UTxPool.
+        Ok(())
+    }
+
+    ///  Insert a set of new transactions into the UTxPool
+    pub fn insert(&mut self, utxs: Vec<Transaction>) -> Result<(), MempoolError> {
+        for utx in utxs.into_iter() {
+            self.insert_utx(utx)?;
+        }
+        Ok(())
+    }
+
+    /// Update the Mempool based on the received published block
+    pub fn process_published_block(&mut self, _published_block: &Block) -> Result<(), MempoolError> {
+        // reorg_pool.insert(utx_pool.remove_published(published_block)?)?; // move published tx to ReOrgPool
+        // orphan_pool.insert(utx_pool.remove_orphaned(published_block)?)?; // The UTXOs of the utx was spent by this
+        // block
+
+        // utx_pool.insert(pending_pool.remove_unconfirmed()?)?; // Utxs with valid input UTXOs and expired time-locks
+        // orphan_pool.insert(pending_pool.remove_orphaned()?)?; // UTXOs of the time-locked utx was spent by this block
+
+        // pending_pool.insert(orphan_pool.remove_time_locked()?)?; // Time-locked Utxs with valid input UTXOs
+        // utx_pool.insert(orphan_pool.remove_unconfirmed()?)?; // Utxs with valid input UTXOs and no time-locks
+
+        // UTx and Txs stored in the OrphanPool and ReOrg pool will be removed when their TTLs are reached.
+
+        Ok(())
+    }
+
+    /// Update the Mempool based on the received set of published blocks
+    pub fn process_published_blocks(&mut self, published_blocks: &Vec<Block>) -> Result<(), MempoolError> {
+        for published_block in published_blocks {
+            self.process_published_block(published_block)?;
+        }
+        Ok(())
+    }
+
+    /// In the event of a ReOrg, resubmit all ReOrged transactions into the Mempool and process each newly introduced
+    /// block from the latest longest chain
+    pub fn process_reorg(&mut self, _removed_blocks: Vec<Block>, _new_blocks: Vec<Block>) -> Result<(), MempoolError> {
+        // self.insert(reorg_pool.remove_unconfirmed(removed_blocks)?)?;
+        // self.process_published_blocks(new_blocks)?;
+
+        Ok(())
+    }
+
+    /// Returns all unconfirmed transaction stored in the Mempool, except the transactions stored in the ReOrgPool.
+    pub fn snapshot(&self) -> Result<Vec<Transaction>, MempoolError> {
+        // return content of UTxPool, OrphanPool and PendingPool, could maybe be returned using future streams
+
+        Ok(Vec::new())
+    }
+
+    /// Returns a list of transaction ranked by transaction priority.
+    pub fn retrieve(&self) -> Result<Vec<Transaction>, MempoolError> {
+        Ok(Vec::new())
+    }
+
+    /// Check if the specified transaction is stored in the Mempool.
+    pub fn contains_key(&self, tx_excess_sig: &Signature) -> Result<(), MempoolError> {
```

Comment:
I think the method should be more specific about how it's checking for a key. Perhaps something like:
```suggestion
    pub fn has_tx_with_excess_signature(&self, sig: &Signature) -> Result<(), MempoolError> {
```

Reply:
Changed to `pub fn has_tx_with_excess_sig(&self, excess_sig: &Signature)` as the name is specified as `excess_sig` in the kernel of transaction.